### ISSUE TITLE
Make configs to be read-only after flags-from-args update

### ISSUE THREAD

**Issue by jdaaph** [PhDSelfHelp/mlmono#7]
It'd be nice to have this function:

JX do you think this is a good starting point for you to get familiar with Python OOP? What I propose is definitely not the only way to do this tho...

two steps:
1. original config is a modifiable class instance that can be updated from passed args, see Issue #4 , you can implement this override by using arg-parse to read in the later string contains all the multi-level flags.

2. you can probably make a subclass of config that basically programmatically iterate over all field recursively and set them as python property (maybe see https://www.python-course.eu/python3_properties.php)

**Comment by jdaaph** [comment]
Also this would ensure future development safety by avoiding accidental modification of config by any class

**Comment by jxwulittlebean** [comment]
Sure! Thanks Diu!!
